### PR TITLE
frontends.torch.dsplit

### LINKED ISSUE #10146: dsplit

### PR DESCRIPTION
**Pull request by xoiga123**
Close #10146 

1. Add `frontends.torch.dsplit` and `test_torch_dsplit`.
2. Fixed type hints for `dsplit` in backends.
3. Removed `out` argument from `dsplit` in backends (it returns a list of arrays).
4. Fixed `dsplit` docstring examples.

### CODE REVIEW

**Comment by xoiga123** [comment]
@fspyridakos somehow because I removed `out` from `test_dsplit`, all the failing tests in `experimental/core/manipulation` are triggered. Before I committed 762153d067e78cac78cbe9097cc5732d5393490e, only `test_dsplit` is failing because of `out`, I can rewind the commit for u to check more easily. I don't think we can support out though, I manually tried and it didn't work.

File: ivy/functional/backends/torch/experimental/manipulation.py [hunk 1]
```diff
@@ -156,11 +156,9 @@ def vsplit(
...
 def dsplit(
     ary: torch.Tensor,
-    indices_or_sections: Union[int, Tuple[int]],
+    indices_or_sections: Union[int, Tuple[int, ...]],
     /,
-    *,
-    out: Optional[torch.Tensor] = None,
-) -> torch.Tensor:
+) -> Tuple[torch.Tensor, ...]:
     return torch.dsplit(ary, indices_or_sections)
```
Comment (fspyridakos) [inline ivy/functional/backends/torch/experimental/manipulation.py:162]: Output should be converted to list to align with the rest of the backends

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Many thanks for your contribution! Looks pretty good apart from my one comment. Probably should rename the output typestrings to `List` instead of `Sequence`

File: ivy/functional/backends/torch/experimental/manipulation.py (hunk 1 above)
Comment (xoiga123) [reply ivy/functional/backends/torch/experimental/manipulation.py]: Thanks, updated it

**Pull request merged**